Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/cart/CartViewModel.kt

```diff
@@ -27,18 +30,26 @@ import kotlinx.coroutines.flow.StateFlow
  *
  * TODO: Move data to Repository so it can be displayed and changed consistently throughout the app.
  */
-class CartViewModel : ViewModel() {
+class CartViewModel(
+    private val snackbarManager: SnackbarManager,
+    snackRepository: SnackRepo
+) : ViewModel() {
+
     private val _orderLines: MutableStateFlow<List<OrderLine>> =
-        MutableStateFlow(SnackRepo.getCart())
+        MutableStateFlow(snackRepository.getCart())
     val orderLines: StateFlow<List<OrderLine>> get() = _orderLines
 
-    fun removeSnack(snackId: Long) {
-        _orderLines.value = _orderLines.value.filter { it.snack.id != snackId }
-    }
+    private var increaseCountErrorCounter = 0
 
     fun increaseSnackCount(snackId: Long) {
-        val currentCount = _orderLines.value.first { it.snack.id == snackId }.count
-        updateSnackCount(snackId, currentCount + 1)
+        if (increaseCountErrorCounter < 4) {
+            val currentCount = _orderLines.value.first { it.snack.id == snackId }.count
+            updateSnackCount(snackId, currentCount + 1)
+            increaseCountErrorCounter++
+        } else {
+            increaseCountErrorCounter = 0
+            snackbarManager.showMessage(R.string.cart_increase_error)
+        }
```

Comment:
Can you use something similar to Jetnews failure counter? 
```
private fun shouldRandomlyFail(): Boolean = ++requestCount % 5 == 0
```

Reply:
Improved logic, thanks!